src/openforms/prefill/sources.py

```diff
@@ -70,7 +70,12 @@ def invoke_plugin(
             if values:
                 logevent.prefill_retrieve_success(submission, plugin, fields)
             else:
-                logevent.prefill_retrieve_empty(submission, plugin, fields)
+                if (
+                    plugin.requires_auth is None
+                    or (auth_info := getattr(submission, "auth_info", None))
+                    and auth_info.attribute == plugin.requires_auth
```

Comment:
```suggestion
                    (required_attribute := plugin.requires_auth) is None
                    or (
                        (auth_info := getattr(submission, "auth_info", None))
                        and auth_info.attribute == required_attribute
                   )
```

* the walrus operator on `plugin.requires_auth` makes it easier to interpret the last condition
* the `and` has precedence, but I prefer wrapping this for readability